internal/npm_install/generate_build_file.js

```diff
@@ -518,9 +542,8 @@ function findScopes() {
 function parsePackage(p) {
   // Parse the package.json file of this package
   const packageJson = path.posix.join(p, 'package.json');
-  const pkg = isFile(packageJson) ?
-      JSON.parse(fs.readFileSync(`${p}/package.json`, {encoding: 'utf8'})) :
-      {version: '0.0.0'};
+  const pkg = isFile(packageJson) ? JSON.parse(fs.readFileSync(packageJson, {encoding: 'utf8'})) :
```

Comment:
Suggestion for a follow up fix:
The `package` object has gotten complex enough that I think it should be a Class, especially since this is Javascript, it's hard to know what custom properties are added to the object later on.

Reply:
Yea. I agree. This file has grown organically and needs refactoring. But I think the additional diff would be too great for this PR so we should refactor to classes & typescript in a follow up.

Reply:
Yes, I meant to say that the refactoring should be considered in a follow up PR :)

Reply:
Also to consider is the multiple filesystem accesses being done on this line (looks like 3 total).  Going from user to kernel space (and back again) is expensive.  Catching exceptions on the `readFileSync` call would be the most efficient in this regard and also prevents the possibility of an error due to changes in the filesystem between the calls.